does not build on current Tesseract anymore

On current tesserocr:master and tesseract:main with Python 3.8, we now got:
```
pip3 install ./repo/tesserocr
Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com
Processing ./repo/tesserocr
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: tesserocr
  Building wheel for tesserocr (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for tesserocr (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [25 lines of output]
      Supporting tesseract v5.3.4
      Tesseract major version 5
      Configs from pkg-config: {'library_dirs': ['/ocrd_all/venv38/lib', '/usr/local/lib'], 'include_dirs': ['/venv38/include', '/usr/local/include', '/usr/local/include'], 'libraries': ['tesseract', 'archive', 'curl', 'lept'], 'compile_time_env': {'TESSERACT_MAJOR_VERSION': 5, 'TESSERACT_VERSION': 84083712}}
      running bdist_wheel
      running build
      running build_ext
      Detected compiler: unix
      building 'tesserocr' extension
      creating build
      creating build/temp.linux-x86_64-cpython-38
      x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/venv38/include -I/usr/local/include -I/usr/local/include -I/venv38/include -I/usr/include/python3.8 -c tesserocr.cpp -o build/temp.linux-x86_64-cpython-38/tesserocr.o -std=c++11 -DUSE_STD_NAMESPACE
      In file included from tesserocr.cpp:1274:0:
      /venv38/include/tesseract/renderer.h:143:24: error: ‘AppendData’ function uses ‘auto’ type specifier without trailing return type
         auto AppendData(T &&d) {
                              ^
     /venv38/include/tesseract/renderer.h:143:24: note: deduced return type only available with -std=c++14 or -std=gnu++14
      tesserocr.cpp: In function ‘PyObject* __pyx_pf_9tesserocr_13PyTessBaseAPI_34GetAvailableLanguages(__pyx_obj_9tesserocr_PyTessBaseAPI*)’:
      tesserocr.cpp:26215:35: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
           for (__pyx_t_4 = 0; __pyx_t_4 < __pyx_t_3; __pyx_t_4+=1) {
                               ~~~~~~~~~~^~~~~~~~~~~
      tesserocr.cpp: In function ‘PyObject* __pyx_pf_9tesserocr_12get_languages(PyObject*, PyObject*)’:
      tesserocr.cpp:40045:35: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
           for (__pyx_t_5 = 0; __pyx_t_5 < __pyx_t_4; __pyx_t_5+=1) {
                               ~~~~~~~~~~^~~~~~~~~~~
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for tesserocr
Failed to build tesserocr
```

full log: see [this build](https://app.circleci.com/pipelines/github/OCR-D/ocrd_all/1028/workflows/bfc969b9-0824-4067-a593-3edd6c1737f5/jobs/1395?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary)

I believe, the reason is that Tesseract now requires C++17. Will investigate...

--- a/setup.py
+++ b/setup.py
@@ -32,6 +32,10 @@ EXTRA_COMPILE_ARGS = {
     'msvc': ['/std:c11', '-DUSE_STD_NAMESPACE'],
     'gcc': ['-std=c++11', '-DUSE_STD_NAMESPACE'],
 }
+EXTRA_COMPILE_ARGS5 = {
+    'msvc': ['/std:c17', '-DUSE_STD_NAMESPACE'],
+    'gcc': ['-std=c++17', '-DUSE_STD_NAMESPACE'],
+}
 
 
 def read(*parts):
@@ -269,13 +273,17 @@ class my_build_ext(build_ext, object):
         compiler = self.compiler.compiler_type
         _LOGGER.info('Detected compiler: %s', compiler)
         extra_args = EXTRA_COMPILE_ARGS.get(compiler, EXTRA_COMPILE_ARGS['gcc'])
+        extra_args5 = EXTRA_COMPILE_ARGS5.get(compiler, EXTRA_COMPILE_ARGS5['gcc'])
         if isinstance(_CYTHON_COMPILE_TIME_ENV, dict):
             version = _CYTHON_COMPILE_TIME_ENV.get('TESSERACT_VERSION', 0)
         else:
             version = 0
 
         for extension in self.extensions:
-            if version >= 0x3050200:
+            if version >= 0x5030400:
+                _LOGGER.debug('tesseract >= 05.03.04 requires c++17 compiler support')
+                extension.extra_compile_args = extra_args5
+            elif version >= 0x3050200:
                 _LOGGER.debug('tesseract >= 03.05.02 requires c++11 compiler support')
                 extension.extra_compile_args = extra_args
 

